### ISSUE TITLE
Programmatic notifications

### ISSUE THREAD

**Issue by tsipinakis** [rfjakob/earlyoom#254]
## Feature proposal

While earlyoom already has a notification system, it seems geared towards desktop notifications and not as much to responding programmatically to oom kill events. I suggest is to implement a script option where all information earlyoom has about the process is passed via arguments environment variables.
e.g.

```
earlyoom --script /etc/earlyoom/notify.sh
```
In the script the following variables can be available:
```
KILLED_UID
KILLED_GID
KILLED_PID
KILLED_NAME
...
```

In this case any other reaction to earlyoom triggers can be implemented with this - it should cover a large number of use-cases.

This is a more general take on #237 .

## Usecase

We are currently trialing using earlyoom to improve the overall health and responsiveness of the interactive user services in our datacenter. Since this is a service users actively work on we'd like to notify via various means (e.g. email) the users to explain what happened/why their application was killed.

The initial idea was to use the existing notification system to write a daemon to forward that information, however the notifications lack the uid/username and group to be able to route the notifications accordingly, so this proposal was created.

I thought it's better to ask your opinion here so we can hopefully end with a general feature that can be upstreamed, rather than a kludge that will remain forever as a patch. Any alternative ideas satisfying this use-case are also welcome.

**Comment by deliciouslytyped** [comment]
Apparently earlyoof used to have a notification command option, which got swapped out for the dbus thing.

**Comment by rfjakob** [comment]
That is correct (commit 17b3d5ec5e13904f62051d28076a881c4bcc4ad6)

**Comment by rfjakob** [comment]
@tsipinakis Are you up to write this feature? I would accept it.

I think the patch should:

* Resurrect the `-N` flag
* Add a function `notify_ext()` similar to notify()
* Environment variables should have a `EARLYOOM_` prefix

**Comment by tsipinakis** [comment]
Implemented in #256 .

**Comment by rfjakob** [comment]
An example notify script would be appreciated when you have time!
